snmp/tests/test_check.py

```diff
@@ -785,7 +812,7 @@ def test_discovery(aggregator):
     try:
         for _ in range(30):
             check.check(instance)
-            if len(aggregator.metric_names) > 1:
+            if 'snmp.IAmAGauge32' in aggregator.metric_names:
```

Comment:
Note for reviewers: This is more reliable than counting the number of metrics. We now have at lest 4 metrics, instead of 1 previously.